tests/cpp/data/test_array_interface.cu

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright 2021 by Contributors
+ */
+#include <gtest/gtest.h>
+#include <xgboost/host_device_vector.h>
+#include "../helpers.h"
+#include "../../../src/data/array_interface.h"
+
+namespace xgboost {
+
+__global__ void SleepForTest(uint64_t *out, uint64_t duration) {
+  auto start = clock64();
+  auto t = 0;
+  while (t < duration) {
+    t = clock64() - start;
+  }
+  out[0] = t;
+}
+
+TEST(ArrayInterface, Stream) {
+  size_t constexpr kRows = 10, kCols = 10;
+  HostDeviceVector<float> storage;
+  auto arr_str = RandomDataGenerator{kRows, kCols, 0}.GenerateArrayInterface(&storage);
+
+  cudaStream_t stream;
+  cudaStreamCreate(&stream);
+
+  auto j_arr =Json::Load(StringView{arr_str});
+  j_arr["stream"] = Integer(reinterpret_cast<int64_t>(stream));
```

Comment:
Is this a safe cast?

Reply:
When you start reinterpreting bits,it's hard to argue what's safe ..

Reply:
Is the stream a memory address, or is it something else, like a struct?

Reply:
It's an opaque pointer.

Reply:
Got it.